### PR TITLE
WIP: Support Oracle Service Names as well as SIDs

### PR DESCRIPTION
**Pull request by negast**

### CODE REVIEW

**Comment by negast** [comment]
I added an additonal param called oracle_SID which is set to true by default. If set to true the old syntax is used else the service syntax is used. I added this param to the readme documentation.

File: manifests/init.pp [hunk 1]
```diff
@@ -58,6 +58,7 @@
   $dbpassword                                                       = 'mypassword',
   $dbserver                                                         = 'localhost',
   $dbname                                                           = 'jira',
+  Boolean $oracle_sid                                                  = true,
```
Comment (bastelfreak) [inline manifests/init.pp]: can you please update the alignment of `=` to match the others?

Comment (negast) [reply manifests/init.pp]: Ok done

**Comment by negast** [comment]
I don't get why the build is failing did I do something wrong here?

**Comment by alexjfisher** [comment]
@negast Beaker tests can be a bit flakey. I restarted your build and it passed.

File: manifests/init.pp [hunk 2]
```diff
@@ -231,13 +232,24 @@
     $dburl_real = $dburl
   }
   else {
-    $dburl_real = $db ? {
-      'postgresql' => "jdbc:${db}://${dbserver}:${dbport_real}/${dbname}",
-      'mysql'      => "jdbc:${db}://${dbserver}:${dbport_real}/${dbname}?useUnicode=true&amp;characterEncoding=UTF8&amp;sessionVariables=default_storage_engine=InnoDB",
-      'oracle'     => "jdbc:${db}:thin:@${dbserver}:${dbport_real}:${dbname}",
-      'sqlserver'  => "jdbc:jtds:${db}://${dbserver}:${dbport_real}/${dbname}",
-      'h2'         => "jdbc:h2:file:/${jira::homedir}/database/${dbname}",
+    if oracle_sid {
```
Comment (alexjfisher) [inline manifests/init.pp]: How are these tests passing though?  Isn't this missing a leading `$`??

Also, is there a way of implementing this with less repetition?  Maybe by doing something like...
```puppet
$oracle_jdbc_separator = bool2str($oracle_sid, ':', '/')
```

File: README.md [hunk 3]
```diff
@@ -271,6 +271,12 @@ Set the schema
 
 The Default value is 'public'
 
+##### `$oracle_sid`
```
Comment (alexjfisher) [inline README.md]: Could we change this parameter name?  `oracle_sid` doesn't sound like a Boolean option.  Maybe `oracle_use_sid`??

Comment (alexjfisher) [reply README.md]: Naming things is hard! ;)

File: manifests/init.pp (hunk 2 above)
Comment (negast) [reply manifests/init.pp]: Hmm yeah that does seem weird. I changed it to use the function as you suggested.

File: README.md (hunk 3 above)
Comment (negast) [reply README.md]: I renamed it to the suggested param.

**Comment by bastelfreak** [comment]
@negast can you take a look at the used email address in the commits? It isn't associated with your github account.

**Comment by negast** [comment]
> @negast can you take a look at the used email address in the commits? It isn't associated with your github account.

I'm sorry where can I view this? I also changed my mail address recently for github.

**Comment by bastelfreak** [comment]
The email address is correct now. You can see your avatar next to the commit, which means that you use a email address that is linked to a github account. Otherwise you can see the address with `git show $commitid`

**Comment by oranenj** [comment]
I redid this in #365 , including credit to @negast. I'll close this PR

**Pull request closed without merging**